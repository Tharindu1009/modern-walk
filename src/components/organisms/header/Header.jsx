import './styles/header.scss';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import Label from '../../atoms/label/Label';
import { labelTypes } from '../../../constants';

function Header() {
    const navigate = useNavigate();

    const [labelType] = useState(labelTypes.header);
    const [titleText] = useState("Modern Walk");

    const goToHome = () => {
        // redirect to the home page
        navigate("/");
    }

    return (
        <div data-testid="header" className='header'>
            <div data-testid="headerTitle" className='title' onClick={goToHome}>
                <Label type={labelType} text={titleText}/>
            </div>
        </div>
    )
}

export default Header;