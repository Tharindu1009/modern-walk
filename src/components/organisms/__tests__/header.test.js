import { render, cleanup, fireEvent } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Header from "../header/Header";

jest.mock('react-router-dom', () => ({
    useNavigate: jest.fn(),
}));

describe('Header', () => {
    afterEach(() => {
        cleanup();
    })

    const mockGoToHome = jest.fn();

    it('should render header component correctly', () => {
        const { getByTestId } = render(<Header />);
        const headerElement = getByTestId('header');
        expect(headerElement).toBeInTheDocument();
        expect(headerElement).toHaveTextContent('Modern Walk');
    });

    it('matches header snapshot', () => {
        const tree = TestRenderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});