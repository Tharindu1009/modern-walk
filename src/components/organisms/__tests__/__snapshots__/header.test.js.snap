// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches header snapshot 1`] = `
<div
  className="header"
  data-testid="header"
>
  <div
    className="title"
    data-testid="headerTitle"
    onClick={[Function]}
  >
    <div
      className="label label__header"
      data-testid="label"
    >
       
      Modern Walk
    </div>
  </div>
</div>
`;
